tests/__init__.py

```diff
@@ -0,0 +1,27 @@
+import typing_extensions
+import types
+
+
+def assert_type(actual_object, expected):
+    # rough check whether the types might match
+
+    actual = type(actual_object).__name__.split(".")[-1]
+    error = False
+
+    if isinstance(expected, types.GenericAlias):  # type: ignore[attr-defined]
+        expected = expected.__name__
+
+    if isinstance(expected, str):
+        actual = actual.lower()
+        expected = expected.lower()
+        error = actual not in expected
+    elif expected is None:
+        error = actual_object is not None
+    else:
+        error = not isinstance(actual_object, expected)
+
+    if error:
+        raise TypeError(f"Expected '{expected}' got '{actual}'")
+
+
+typing_extensions.assert_type = assert_type
```

Comment:
A bit ugly and only a rough check (if the actual object is of a sub-class, the caller might need to pass the type and not the name of the class) but it might make the test more concise.

Reply:
I don't understand why you're overriding the default implementation

Reply:
Ideally, we should have test like this:
```py
assert_type(pd.DataFrame.__hash__, None)
assert pd.DataFrame.__hash__ is None
```
Replacing `assert_type` at runtime with a rudimentary check, helps us to keep the tests simpler.

Reply:
Happy to remove it, just thought it might be helpful (even though it is an ugly hack)

Reply:
> Happy to remove it, just thought it might be helpful (even though it is an ugly hack)

I just don't see the benefit and I'd rather not be hacking things in libraries.

The goal of the tests is two-fold:
1. Test with `mypy` and `pyright` to see that "user" code would pass the type checkers
2. Test with `pytest` to see that it is valid pandas code that is runnable.

`assert_type()` at runtime just returns the first argument.  Based on my reading of the docs, I think you could do something like:
```python
assert_type(pd.DataFrame.__hash__, None) is None
```